libs/platform/src/lib/components/list/list.component.ts

```diff
@@ -0,0 +1,631 @@
+                item.navigated = this._navigated;
+                item.navigationIndicator = this._navigationIndicator;
+                item.contentDensity = this.contentDensity;
+                item.selectionMode = this.selectionMode;
+                item.listType = this.listType;
+                item.hasByLine = this._hasByLine;
+                item.noSeperator = this.noSeperator;
+            });
+        });
+    }
+
+    ngOnDestroy(): void {
+        if (isDataSource(this.dataSource)) {
+            this.dataSource.close();
+        }
+        if (this._dsSubscription) {
+            this._dsSubscription.unsubscribe();
+        }
+        if (this.selectionModel) {
+            this.selectionModel.changed.unsubscribe();
+        }
+        if (this._itemsSubscription) {
+            this._itemsSubscription.unsubscribe();
+        }
+    }
+
+    get value(): any {
+        return this.selectedItems;
+    }
+    set value(value: any) {
+        this.selectedItems = value;
+        this._changeDetectorRef.markForCheck();
+    }
+
+
+    /** @hidden */
+    /**handline keyboard operations
+    *  on list and list items
+    */
+    handleKeyDown(event: any): boolean {
+        event.stopImmediatePropagation();
+        if (this.keyManager) {
+            if (event.keyCode === DOWN_ARROW || event.keyCode === UP_ARROW) {
+                return false;
+            } else if (event.keyCode === ENTER || event.keyCode === SPACE) {
+                this.updateNavigation(event);
+                return false;
+            }
+        }
+    }
+
+    private initializeDS(ds: FdpListDataSource<any>): void {
+        this.dsItems = [];
+        if (isDataSource(this.dataSource)) {
+            this.dataSource.close();
+            if (this._dsSubscription) {
+                this._dsSubscription.unsubscribe();
+                this._dsSubscription = null;
+            }
+        }
+        // Convert ListDataSource<T> | T[] as DataSource
+        this._dataSource = this.openDataStream(ds);
+    }
+
+    private toDataStream(ds: FdpListDataSource<any>): ListDataSource<any> {
+        if (isDataSource(ds)) {
+            return ds;
+        }
+        return undefined;
+    }
+
+    private openDataStream(ds: FdpListDataSource<any>): ListDataSource<any> {
+        const initDataSource = this.toDataStream(ds);
+        if (initDataSource === undefined) {
+            throw new Error(`[dataSource] source did not match an array, Observable, or DataSource`);
+        }
+        /**
+         * This is single point of data entry to the component. We dont want to set data on different
+         * places. If any new data comes in either you do a search and you want to pass initial data
+         * its here.
+         */
+        this._dsSubscription = initDataSource
+            .open()
+            .pipe(takeUntil(this._destroyed))
+            .subscribe((data) => {
+                this.dsItems = data || [];
+                this._cd.markForCheck();
```

Comment:
emit event to parent with stateChanges. We need to keep in mind that all this is wrapped with FormField and this component is responsible for refresh the view.

Reply:
updated here and in all applicable places for list.componen.ts